apps/alert_processor/lib/model/query.ex

```diff
@@ -82,6 +86,58 @@ defmodule AlertProcessor.Model.Query do
         LEFT JOIN subscriptions s ON u.id = s.user_id
         GROUP BY u.communication_mode
         """
+      },
+      %__MODULE__{
+        label: "Abandonment rate in the past 500 weeks",
+        query: """
+        WITH
+          periods AS (
+            SELECT \"end\" - interval '1 week' AS start, \"end\"
+            FROM generate_series(
+              #{@nearest_weekly_period_end} - interval '500 weeks',
```

Comment:
Instead of a fixed interval, can we put the result of `SELECT MIN(inserted_at) FROM users` here? That way it will always go back for as long as we have data, and also won't go any further than it needs to (testing this on production right now, there's a whole bunch of extra rows at the end that are all zeros).